assets/javascripts/discourse/initializers/add-noindex-item-to-topic-wrench-menu.js

```diff
@@ -0,0 +1,50 @@
+import { withPluginApi } from "discourse/lib/plugin-api";
+import { ajax } from "discourse/lib/ajax";
+
+// eslint-disable-next-line no-unused-vars
+const PLUGIN_ID = "discourse-topic-noindex";
+
+export default {
+  name: "discourse-topic-noindex",
+
+  initialize() {
+    withPluginApi("0.8.31", initialize);
+  },
+};
+
+function initialize(api) {
+  let helper = null;
+
+  api.attachWidgetAction(
+    "topic-admin-menu:adminMenuButtons",
+    "toggleNoIndex",
+    () => {
+      const topic = helper?.attrs?.topic;
+      if (!topic) {
+        return;
+      }
+      ajax(`/t/${topic.id}/toggle-noindex`, {
+        type: "PUT",
+      }).then(() => {
+        topic.reload();
+      });
+    }
+  );
+
+  api.decorateWidget("topic-admin-menu:adminMenuButtons", (_helper) => {
+    helper = _helper;
```

Comment:
Any reason to rename this variable? Couldn't it just be

```suggestion
  api.decorateWidget("topic-admin-menu:adminMenuButtons", (helper) => {
```